index.html

```diff
@@ -1688,40 +1709,75 @@
             </dd>
           </dl>
           <ol>
-            <li>If the <a>presentation connection state</a> of
-            <code>presentationConnection</code> is not <code>connected</code>
-            or <code>connecting</code>, then abort these steps.
-            </li>
-            <li>Set <a>presentation connection state</a> of
+            <li>Start to tear down the underlying connection in a non abrupt
+            manner. The mechanism by which this is achieved is
+            implementation-specific but should signal the intention to tear
+            down the connection to the <a>destination browsing context</a> and
+            eventually render the connection <a link-for=
+            "PresentationConnection">closed</a>.
+            </li>
+            <li>If the procedure was initiated by the <code><a link-for=
+            "PresentationConnection">close</a>()</code> method, then set the
+            <a>presentation connection state</a> of
             <code>presentationConnection</code> to <code>closed</code>.
             </li>
-            <li>
-              <a>Queue a task</a> to <a>fire</a> a <a>trusted event</a> with
-              the name <code>closed</code>, that uses the
-              <a>PresentationConnectionClosedEvent</a> interface, with the
-              <a for="PresentationConnectionClosedEvent">reason</a> attribute
-              initialized to <var>closeReason</var> and the <a for=
-              "PresentationConnectionClosedEvent">message</a> attribute
-              initialized to <var>closeMessage</var>, at
-              <code>presentationConnection</code>. The event must not bubble,
-              must not be cancelable, and has no default action.
+            <li>Otherwise, <a>queue a task</a> to set the <a>presentation
+            connection state</a> of <code>presentationConnection</code> to
+            <code>closed</code>.
```

Comment:
It seems like setting the state to closed and firing the CloseEvent should happen in the same run of the event loop?